src/components/plots/plot.ts

```diff
@@ -514,27 +486,16 @@ module Plottable {
     /**
      * Retrieves all of the PlotData of this plot for the specified dataset(s)
      *
-     * @param {string | string[]} datasetKeys The dataset(s) to retrieve the selections from.
-     * If not provided, all selections will be retrieved.
+     * @param {Dataset[]} datasets The Datasets to retrieve the PlotData from.
+     * If not provided, all PlotData will be retrieved.
      * @returns {PlotData} The retrieved PlotData.
      */
-    public getAllPlotData(datasetKeys: string | string[] = this.datasetOrder()): Plots.PlotData {
-      var datasetKeyArray: string[] = [];
-      if (typeof(datasetKeys) === "string") {
-        datasetKeyArray = [<string> datasetKeys];
-      } else {
-        datasetKeyArray = <string[]> datasetKeys;
-      }
-
-      return this._getAllPlotData(datasetKeyArray);
-    }
-
-    protected _getAllPlotData(datasetKeys: string[]): Plots.PlotData {
+    public getAllPlotData(datasets = this.datasets()): Plots.PlotData {
```

Comment:
Why can't we allow for Dataset | Dataset[]?  Our architecture isn't hurt that much and it allows for a nice streamlined API